org.thecrossingchurch.CustomJobs/Jobs/HubspotIntegrationPatching.cs

```diff
@@ -0,0 +1,442 @@
+                }
+            }
+            Debug.WriteLine( string.Format("{0} Emails on ENews List.",eNewsData.Count ) );
+
+
+            HashSet<string> hsPersonEmails = new HashSet<string>();
+            foreach ( var contact in contacts )
+            {
+                if ( contact.properties.email.ToStringSafe() != null )
+                {
+                    hsPersonEmails.Add( contact.properties.email.ToStringSafe().ToLower() );
+                }
+            }
+
+            //WriteToLog( string.Format( "Total Contacts: {0}", contacts.Count() ) );
+            for ( var i = 0; i < contacts.Count(); i++ )
+            {
+                //Stopwatch watch = new Stopwatch();
+                //watch.Start();
+                Person person = personService.Get(contacts[i].properties.rock_person_id);
+
+                //For Testing
+                //WriteToLog( string.Format( "{1}i: {0}{1}", i, Environment.NewLine ) );
+                //WriteToLog( string.Format( "    After SQL: {0}{1}", watch.ElapsedMilliseconds, Environment.NewLine ) );
+
+                // If person is null, that means that we have a person in HS w/ a personId that no longer exists in rock
+                // This implies that a merge occurred and we need to look at the alias table to figure out what the new Id is and update it
+                // After updating the ID, we need to find the person object and handle patching
+
+                // Setup for patching
+                // Look up hubspot defined type
+                DefinedTypeService definedTypeService = new DefinedTypeService( _context );
+                DefinedType hsDefinedType = definedTypeService.Get( 527 ); // as of 7/8/24 Dev is 528, Train is 527
+                AttributeValueService attributeValueService = new AttributeValueService( _context );
+                AttributeService attributeService = new AttributeService( _context );
+
+
+                //Schedule HubSpot update if 1:1 match
+                if ( person != null )
+                {
+                    current_id = person.Id;
+                    var url = $"https://api.hubapi.com/crm/v3/objects/contacts/{contacts[i].id}";
+                    Debug.WriteLine( "Contact Count: " + i + " of " + contacts.Count() );
+                    Debug.WriteLine( "URL: " + url );
+                    var properties = new List<HubspotPropertyUpdate>();
+
+
+                    foreach ( DefinedValue hsSyncDv in hsDefinedType.DefinedValues )
+                    {
+                        hsSyncDv.LoadAttributes();
+                        Dictionary<string, AttributeValueCache> dvAttributes = hsSyncDv.AttributeValues;
+                        
+                        string propertyOrAttribute = dvAttributes.GetValueOrNull("IsPropertyOrAttribute").Value;
+                        string hsKey = dvAttributes.GetValueOrNull("HubSpotAttributeKey").Value;
+                        string type = dvAttributes.GetValueOrNull( "Type" ).Value;
```

Comment:
Can we add null handling to this in case the defined value doesn't have these attribute values?